presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -255,10 +262,12 @@ public Ssl()
     {
         private static final Predicate<Properties> IF_SSL_ENABLED =
                 checkedPredicate(properties -> SSL.getValue(properties).orElse(false));
+        private static final Predicate<Properties> IF_SSL_VERIFICATION_ENABLED =
+                checkedPredicate(properties -> !SSL_VERIFICATION.getValue(properties).orElse(SslVerificationMode.FULL).equals(SslVerificationMode.NONE));
 
         public SslKeyStorePath()
         {
-            super("SSLKeyStorePath", NOT_REQUIRED, IF_SSL_ENABLED, STRING_CONVERTER);
+            super("SSLKeyStorePath", NOT_REQUIRED, IF_SSL_ENABLED.and(IF_SSL_VERIFICATION_ENABLED), STRING_CONVERTER);
```

Comment:
Please correct me if I am wrong. Key stores are not to used to verify SSL but to authenticate using SSL. So SSL verification should not affect this.

Reply:
The current implementation follows the description in https://github.com/prestosql/presto/issues/5246.

When the `SSLVerification` mode is `NONE`, the following properties will be disallowed:

- `SSLKeyStorePath`
- `SSLKeyStorePassword`
- `SSLKeyStoreType`
- `SSLTrustStorePath`
- `SSLTrustStorePassword`
- `SSLTrustStoreType`

Reply:
Ok, as I understand to goal is to make impossible to use SSL (certificate) authentication with `SSLVerification=NONE`. 

@electrum Do I understand this correctly?